src/common.h

```diff
@@ -169,4 +178,84 @@ bool Contains(const std::vector<std::string>& vec, const std::string& str);
+
+#undef TRITON_BIG_ENDIAN
+    
+#define TRITON_BIG_ENDIAN true
+
+/// Converts from host byte order to little endian byte order in place.
+///
+/// If host byte order is little endian then defined as an empty macro.
+///
+/// \param[in] datatype data type of array
+/// \param[in, out] base pointer to array of data type elements to convert
+/// \param[in] byte_size size of array in bytes
+///
+void HostToLittleEndian(
+    TRITONSERVER_DataType datatype, char* base, size_t byte_size);
+
+/// Converts from little endian byte order to host byte order in place.
+///
+/// If host byte order is little endian then defined as an empty macro.
+///
+/// \param[in] datatype data type of array
+/// \param[in, out] base pointer to array of data type elements to convert
+/// \param[in] byte_size size of array in bytes
+///
+void LittleEndianToHost(
+    TRITONSERVER_DataType datatype, char* base, size_t byte_size);
+
+/// Converts from little endian byte order to host byte order in place
+/// with support for non-contiguous arrays. For non-contiguous arrays
+/// the initial call must pass an empty partial result and 0 for the
+/// initial next_offset. Subsequent calls should pass the returned
+/// values without modification.
+///
+/// If host byte order is little endian then defined as an empty macro.
+///
+/// \param[in] datatype data type of array
+/// \param[in, out] base pointer to array of data type elements to convert
+/// \param[in] byte_size size of array in bytes
+/// \param[in, out] partial_result vector to store partial results
+/// \param[in, out] next_offset intermediate value used for storing next offset
+/// for BYTES data type
+///
+void LittleEndianToHost(
+    TRITONSERVER_DataType datatype, char* base, size_t byte_size,
+    std::vector<char*>& partial_result, size_t& next_offset);
+
+/// Returns the data type for an http raw binary input. Returns the
+/// data type of the first input as raw binary inputs can only be used
+/// with models that have a single input.
+///
+/// If host byte order is little endian then defined as an empty macro.
+///
+/// \param[in] server server with requested model
+/// \param[in] model_name name of model
+/// \param[in] model_version version of model
+/// \return data type of first input or TRITONSERVER_TYPE_INVALID on error.
+///
+TRITONSERVER_DataType GetDataTypeForRawInput(
+    TRITONSERVER_Server* server, const std::string& model_name,
+    const int64_t model_version);
+
+#else
+
+#undef TRITON_BIG_ENDIAN
+
+#define HostToLittleEndian(datatype, base, byte_size) \
+  do {                                                \
+  } while (false)
+#define LittleEndianToHost(datatype, base, byte_size, ...) \
+  do {                                                     \
+  } while (false)
+#define GetDataTypeForRawInput(server, model_name, model_version) \
+  do {                                                            \
+  } while (false)
```

Comment:
I was wondering why we need a macro for the little endian case but we use a function for big endian case? I probably need to dig more into the implementation.

Reply:
Essentially wanted to completely eliminate these in the little endian case. For systems that are little endian already there is nothing to do. Using this to reduce the ifdefs at the place of usage. 

if the host type is already LittleEndian - then the macro HostToLittleEndian still does what is intended - but is a no-op.

Happy to discuss other ways.

Reply:
@Tabrizian Since we only use these in a few places - perfectly happy to guard them with `#ifdef TRITON_BIG_ENDIAN` instead of using empty macros if it makes things a bit clearer. I think initially I was hoping to avoid needing to define `TRITON_BIG_ENDIAN` - and kind of liked the way the functions or macros still behaved correctly without guards.

Reply:
Is it possible to move the ifdefs inside these functions? The body of the function would change based on whether `ifdef TRITON_BIG_ENDIAN` is enabled or not. If it is not enabled, it wouldn't do anything.

Reply:
@nv-kmcgill53 - What do you think - is the compiler smart enough to remove the calls completely? I want to avoid any calls in little endian case entirely - so if compiler would remove an "empty function" as @Tabrizian is suggesting - am good with that as well. I'll dig also - but you might know the answer already.

Reply:
These calls get optimized out for anything greater than `Debug`. This is also a standard pattern in industry so I have no problem with the way it's defined. Mostly this pattern is used for defining debug functions for printing/logging info which then get optimized out in the `Release` build. @nnshah1 you might be able to get away with just not defining anything at all after it so that the symbol gets inserted in the preprocessor and then gets translated to no-op. I need to play around with that though.

Reply:
The current way it is handled should be fine resolving.